clients/include/testing_gemv.hpp

```diff
@@ -31,82 +30,81 @@ void testing_gemv_bad_arg()
     rocblas_init<T>(hA, M, N, lda);
     rocblas_init<T>(hx, 1, N, incx);
     rocblas_init<T>(hy, 1, M, incy);
 
     // allocate memory on device
-    auto dA_managed = rocblas_unique_ptr{rocblas_test::device_malloc(sizeof(T) * size_A),
-                                         rocblas_test::device_free};
-    auto dx_managed = rocblas_unique_ptr{rocblas_test::device_malloc(sizeof(T) * size_x),
-                                         rocblas_test::device_free};
-    auto dy_managed = rocblas_unique_ptr{rocblas_test::device_malloc(sizeof(T) * size_y),
-                                         rocblas_test::device_free};
-    T* dA = (T*)dA_managed.get();
-    T* dx = (T*)dx_managed.get();
-    T* dy = (T*)dy_managed.get();
+    device_vector<T> dA(size_A);
+    device_vector<T> dx(size_x);
+    device_vector<T> dy(size_y);
     if(!dA || !dx || !dy)
     {
         PRINT_IF_HIP_ERROR(hipErrorOutOfMemory);
         return;
     }
 
     // copy data from CPU to device
-    CHECK_HIP_ERROR(hipMemcpy(dA, hA.data(), sizeof(T) * size_A, hipMemcpyHostToDevice));
-    CHECK_HIP_ERROR(hipMemcpy(dx, hx.data(), sizeof(T) * size_x * incx, hipMemcpyHostToDevice));
-    CHECK_HIP_ERROR(hipMemcpy(dy, hy.data(), sizeof(T) * size_y * incy, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(dA, hA, sizeof(T) * size_A, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(dx, hx, sizeof(T) * size_x, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(dy, hy, sizeof(T) * size_y, hipMemcpyHostToDevice));
 
     {
         T* dA_null = nullptr;
 
-        status = rocblas_gemv<T>(
-            handle, transA, M, N, (T*)&alpha, dA_null, lda, dx, incx, (T*)&beta, dy, incy);
+        status =
+            rocblas_gemv<T>(handle, transA, M, N, &alpha, dA_null, lda, dx, incx, &beta, dy, incy);
 
         verify_rocblas_status_invalid_pointer(status, "rocBLAS TEST ERROR: A is null pointer");
     }
     {
         T* dx_null = nullptr;
-        status     = rocblas_gemv<T>(
-            handle, transA, M, N, (T*)&alpha, dA, lda, dx_null, incx, (T*)&beta, dy, incy);
+        status =
+            rocblas_gemv<T>(handle, transA, M, N, &alpha, dA, lda, dx_null, incx, &beta, dy, incy);
 
         verify_rocblas_status_invalid_pointer(status, "rocBLAS TEST ERROR: x is null pointer");
     }
     {
         T* dy_null = nullptr;
-        status     = rocblas_gemv<T>(
-            handle, transA, M, N, (T*)&alpha, dA, lda, dx, incx, (T*)&beta, dy_null, incy);
+        status =
+            rocblas_gemv<T>(handle, transA, M, N, &alpha, dA, lda, dx, incx, &beta, dy_null, incy);
 
         verify_rocblas_status_invalid_pointer(status, "rocBLAS TEST ERROR: y is null pointer");
     }
+    {
+        T* alpha_null = nullptr;
+        status =
+            rocblas_gemv<T>(handle, transA, M, N, alpha_null, dA, lda, dx, incx, &beta, dy, incy);
+
+        verify_rocblas_status_invalid_pointer(status, "rocBLAS TEST ERROR: alpha is null pointer");
+    }
```

Comment:
alpha == NULL test was omitted from the GEMV tests, and from the GEMV code itself. This adds it to the tests (an earlier PR added it to the library).